src/js/angular/ttyg/services/ttyg-context.service.js

```diff
@@ -36,6 +36,27 @@ function TTYGContextService(EventEmitterService) {
      */
     let _selectedAgent = undefined;
 
+    /**
+     * The default agent values.
+     * @type {AgentModel|undefined}
+     * @private
+     */
+    let _defaultAgent = undefined;
+
+    /**
+     * @return {Promise<AgentModel>}
+     */
+    const getDefaultAgent = () => {
+        if (_defaultAgent) {
+            return Promise.resolve(cloneDeep(_defaultAgent));
+        }
+        return TTYGService.getDefaultAgent()
```

Comment:
I prefer this to be somewhere else, but not here. Perhaps the ttyg-view.controller is better place. The context service should not deal with loading data

Reply:
Moved it to the `ttyg-view.controller`